pgjdbc/src/main/java/org/postgresql/jdbc/PgResultSet.java

```diff
@@ -1667,7 +1667,7 @@ boolean isUpdateable() throws SQLException {
     /*
     it is only updatable if the primary keys are available in the resultset
      */
-    updateable = (i == numPKcolumns);
+    updateable = (i == numPKcolumns) & (numPKcolumns > 0);
```

Comment:
missed an & here should be &&

Reply:
& also useful here. but && will be a better choice here indeed.